pkg/controller/component/component_controller.go

```diff
@@ -0,0 +1,252 @@
+// and what is in the Component.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileComponent) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Component")
+
+	// Fetch the Component instance
+	instance := &componentsv1alpha1.Component{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	log.Info("============================================================")
+	log.Info(fmt.Sprintf("***** Reconciling Component %s, namespace %s", request.Name, request.Namespace))
+	log.Info(fmt.Sprintf("** Creation time : %s", instance.ObjectMeta.CreationTimestamp))
+	log.Info(fmt.Sprintf("** Resource version : %s", instance.ObjectMeta.ResourceVersion))
+	log.Info(fmt.Sprintf("** Generation version : %s", strconv.FormatInt(instance.ObjectMeta.Generation, 10)))
+	log.Info(fmt.Sprintf("** Deletion time : %s", instance.ObjectMeta.DeletionTimestamp))
+	log.Info("============================================================")
+
+	// Assign the generated ResourceVersion to the resource status
+	if instance.Status.RevNumber == "" {
+		instance.Status.RevNumber = instance.ObjectMeta.ResourceVersion
+	}
+
+	if !instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info( "** DELETION **")
+	}
+
+	// We only call the pipeline when the component has been created
+	// and if the Status Revision Number is the same
+	if instance.Status.RevNumber == instance.ObjectMeta.ResourceVersion {
+		// Create an empty image name "nodejs-output"
+		newImageForOutput := newImageStream(instance.Namespace, instance.Name)
+		err = r.client.Create(context.TODO(), newImageForOutput)
+		if err != nil {
+			log.Error(err, "** Creating new OUTPUT image fails **")
+			return reconcile.Result{}, err
+		}
+		log.Info("** Image stream for OUTPUT created **")
+		if err := controllerutil.SetControllerReference(instance, newImageForOutput, r.scheme); err != nil {
+			log.Error(err, "** Setting owner reference fails **")
+			return reconcile.Result{}, err
+		}
+
+		// Create an empty image name "nodejs-runtime"
+		newImageForRuntime := newImageStreamFromDocker(instance.Namespace, instance.Name, instance.Spec.BuildType)
+		err = r.client.Create(context.TODO(), newImageForRuntime)
```

Comment:
`newImageForRuntime` could be nil. Should it be handled?

Reply:
yes indeed
870bc31

Reply:
@corinnekrych, the fix could still cause problems. 
Can you please split the `if err != nil || newImageForRuntime == nil {` line two two seperate If checks?
That would allow us to have different error logging in both the cases.